File: pantherwatch/src/main/java/edu/gsu/pantherwatch/pantherwatch/api/RetrieveCourseInfoResponse.java
package edu.gsu.pantherwatch.pantherwatch.api;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;

@Data
public class RetrieveCourseInfoResponse {
    @JsonProperty("success")
    private boolean success;

    @JsonProperty("totalCount")
    private int totalCount;

    @JsonProperty("data")
    private CourseData[] data;

    @JsonProperty("pageOffset")
    private int pageOffset;

    @JsonProperty("pageMaxSize")
    private int pageMaxSize;
}

File: frontend-android/app/src/main/java/edu/gsu/pantherwatch/data/model/RetrieveCourseInfoResponse.kt
package edu.gsu.pantherwatch.data.model

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class RetrieveCourseInfoResponse(
    @field:Json(name = "success")
    val success: Boolean,
    
    @field:Json(name = "totalCount")
    val totalCount: Int,
    
    @field:Json(name = "data")
    val data: List<CourseData>,
    
    @field:Json(name = "pageOffset")
    val pageOffset: Int,
    
    @field:Json(name = "pageMaxSize")
    val pageMaxSize: Int
)
